src/main/java/de/picturesafe/search/samples/aggregations/SearchWithTermsAggregation.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.picturesafe.search.samples.aggregations;
+
+import de.picturesafe.search.elasticsearch.DataChangeProcessingMode;
+import de.picturesafe.search.elasticsearch.SingleIndexElasticsearchService;
+import de.picturesafe.search.elasticsearch.model.DocumentBuilder;
+import de.picturesafe.search.elasticsearch.model.ResultFacet;
+import de.picturesafe.search.elasticsearch.model.ResultFacetItem;
+import de.picturesafe.search.elasticsearch.model.SearchResult;
+import de.picturesafe.search.expression.Expression;
+import de.picturesafe.search.expression.FulltextExpression;
+import de.picturesafe.search.parameter.SearchParameter;
+import de.picturesafe.search.parameter.SortOption;
+import de.picturesafe.search.parameter.aggregation.TermsAggregation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Component;
+
+import java.util.stream.LongStream;
+
+@Component
+@ComponentScan
+public class SearchWithTermsAggregation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SearchWithTermsAggregation.class);
+
+    @Autowired
+    private SingleIndexElasticsearchService singleIndexElasticsearchService;
+
+    public static void main(String[] args) {
+        try (AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(SearchWithTermsAggregation.class)) {
+            final SearchWithTermsAggregation facet = ctx.getBean(SearchWithTermsAggregation.class);
+            facet.run();
+        }
+    }
+
+    private void run() {
+        try {
+            singleIndexElasticsearchService.createIndexWithAlias();
+            createTestRecords();
+
+            final Expression expression = new FulltextExpression("test title");
+
+            // Search results and retrieve first page
+            final SearchResult searchResult = singleIndexElasticsearchService.search(expression, createSearchParameter());
+            LOGGER.info("Found {} hits with {} facet(s)", searchResult.getResultCount(), searchResult.getFacets().size());
+            showFacets(searchResult);
+        } finally {
+            singleIndexElasticsearchService.deleteIndexWithAlias();
+        }
+    }
+
+    private void createTestRecords() {
+        // Insert 20 test records with city 'Hamburg'
+        LongStream.range(1, 21)
+                .forEach(id -> singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING,
```

Comment:
A loop with single adds in blocking mode is a serious performance issue! I know this is no production code, but a sample should give a hint on how to use the API correctly.